mdata/notifier.go

```diff
@@ -83,7 +84,10 @@ func (cl Notifier) Handle(data []byte) {
 		}
 
 		// get metric
-		if agg, ok := cl.Metrics.Get(ms.Key); ok {
+		if cl.CreateMissingMetrics {
+			agg := cl.Metrics.GetOrCreate(ms.Key)
```

Comment:
I am not sure what you are saying here.

The reason for the `if cl.CreateMissingMetrics` is that there are two notifier handlers, kafka and NSQ.  NSQ is not partition aware. so it could receive metricPersist messages for metrics this node does not handle. In that case we dont want to automatically create any missing AggMetrics and we dont want to call syncSaveState